bpf/process/types/basic.h

```diff
@@ -2451,6 +2456,10 @@ read_call_arg(void *ctx, struct msg_generic_kprobe *e, int index, int type,
 	case int_type:
 	case s32_ty:
 	case u32_ty:
+	case s16_ty:
+	case u16_ty:
+	case s8_ty:
+	case u8_ty:
```

Comment:
Not sure of the semantics of `probe_read`, but if we read the last byte of a page where the next page is unmapped, I don't see how this wouldn't pagefault. Or am I missing something?

Reply:
hum pagefaults are disabled, so I think it'll return -EFAULT.. but right that's corner case that will fail, I wanted to avoid adding more code in here, but perhaps it's not that horrible

Reply:
I see. I agree that it's a corner case, but part of the problem is that we don't track these errors so there is no way to tell that something went wrong.

Reply:
for what its worth (obvious?) you'll also get EFAULT from reading user memory on a syscall. I don't think we have an error counter there either.

Reply:
ok, I think we could have also some indication/flag that read fail for the event.. and some metric counter

Reply:
> ok, I think we could have also some indication/flag that read fail for the event.. and some metric counter

Planned as followup I presume?

Reply:
yes, todo updated ;-)